src/txdb.cpp

```diff
@@ -448,6 +450,47 @@ bool CBlockTreeDB::WriteTxIndex(const std::vector<std::pair<uint256, CTxIndexVal
     return WriteBatch(batch);
 }
 
+bool CBlockTreeDB::ReadMaturityHeightIndex(const int height, std::vector<CMaturityHeightKey> &val) {
+    boost::scoped_ptr<leveldb::Iterator> pcursor(NewIterator());
+
+    CDataStream ssKeySet(SER_DISK, CLIENT_VERSION);
+    ssKeySet << make_pair(DB_MATURITY_HEIGHT, CMaturityHeightIteratorKey(height));
+    pcursor->Seek(ssKeySet.str());
+
+    while (pcursor->Valid()) {
+        boost::this_thread::interruption_point();
+        try {
+            leveldb::Slice slKey = pcursor->key();
+            CDataStream ssKey(slKey.data(), slKey.data()+slKey.size(), SER_DISK, CLIENT_VERSION);
+            char chType;
+            CMaturityHeightKey indexKey;
+            ssKey >> chType;
+            ssKey >> indexKey;
+            if (chType == DB_MATURITY_HEIGHT && indexKey.blockHeight == height) {
+                val.push_back(indexKey);
+                pcursor->Next();
+            } else {
+                break;
+            }
+        } catch (const std::exception& e) {
+            break;
+        }
+    }
+
+    return true;
+}
+
+bool CBlockTreeDB::UpdateMaturityHeightIndex(const std::vector<std::pair<CMaturityHeightKey,CMaturityHeightValue>> &vect) {
+    CLevelDBBatch batch;
+    for (std::vector<std::pair<CMaturityHeightKey,CMaturityHeightValue> >::const_iterator it=vect.begin(); it!=vect.end(); it++)
+        if (it->second.IsNull()) {
```

Comment:
Add comment to specify this is convention we decided -> `If the value is null we mean we want to erase the pair from the DB otherwise we persist it`